### ISSUE TITLE
Provide a NPM package or SRI signatures

### ISSUE THREAD

**Issue by romainneutron** [recurly/recurly-js#255]
This ticket is a followup of https://github.com/recurly/recurly-js/issues/227. I have a few comments based on your answer:

TLDR: Providing the lib from CDN does not provide more trust than a custom library hosting at the moment. NPM package could provide much more security for end users. Either providing an NPM package or SRI signatures could lead to trustworthy library usage.

> I recommend only depending on Recurly.js from our CDN directly -- this will ensure you always have the latest version, and will bolster your security by trusting our source instead of your own packages.

This is indeed a great thing. However, I do trust much more my own CDN hosting (using [SRI](https://www.w3.org/TR/SRI/)) than an external hosting. You may say it's a point of view, but which guarantee do you provide concerning your CDN compromission?
it would be great if you could provide [SRI](https://www.w3.org/TR/SRI/) so that user agents could check the integrity of the javascript code you provide.

> Most importantly, we are likely to introduce breaking API changes in the future that old versions may not support. By loading Recurly.js from our CDN, you can be assured that you are always compatible.

This is - again - my own point of view: If you break you API and think that it will be transparent for users because web apps loads javascript from your CDN then you're facing a big issue: think about a single page application that loads a script once and can use it for days. Breaking your API will break its usage for all users of this app. 
I'm pretty sure there are way more possibilities like providing BC, and when it's not possible, bump the API version (Github is quite good on this for instance)

Moreover, if you would like to provide SRI support to get more trust for the user agents downloading the JS library from your CDN, you'll have to provide a URL and a SRI signature per version.

**Comment by chrissrogers** [comment]
I agree with your assessment on adding an SRI signature. We'll look into generating these for the current and all ongoing versions.

There are some risks in using recurly.js as an npm dependency specifically related to PCI compliance -- hosting a library which handles customer payment information, even indirectly, may require a higher level of scrutiny in your PCI compliance. But, this will become a non-issue once we publish SRI signatures :)

**Comment by chrissrogers** [comment]
Just a note: if you do specify an integrity hash, it will fail when the recurly.js resource is updated. You would need to monitor our releases (we prepare a pre-release at least 24 hours before releasing a new version) and build for a failover period that incorporates both signatures.

for example

```html
<script ... integrity="hash-v4.0.3">
<script ... integrity="hash-v4.0.4">
```

**Comment by chrissrogers** [comment]
SRI sigs are in the latest release note and will be provided in the future.
